credentials/credentials.go

```diff
@@ -74,6 +75,9 @@ type TransportAuthenticator interface {
 	// Dial connects to the given network address and does the authentication
 	// handshake specified by the corresponding authentication protocol.
 	Dial(addr string) (net.Conn, error)
+	// DialTimeout acts like Dial but takes a timeout. The timeout applies to
+	// connection and handshake as a whole.
+	DialTimeout(addr string, timeout time.Duration) (net.Conn, error)
```

Comment:
you do not need to add a new user-facing API. You can add some logic into Dial instead.